tests/unit/checkboxradio/checkboxradio_options.js

```diff
@@ -0,0 +1,180 @@
+-/*
+ * checkboxradio_methods.js
+ */
+
+
+(function($) {
+
+module( "Checkboxradio: checkbox: options" );
+	test( "options: disabled", function() {
+		var checkbox = $( "#checkbox-option-disabled" ),
+			widget;
+		expect( 10 );
+		checkbox.checkboxradio({
+			disabled: true
+		});
+
+		widget = checkbox.checkboxradio( "widget" );
+
+		strictEqual( widget.hasClass( "ui-state-disabled" ), true,
+			"label gets ui-state-disabled when initial option set to true" );
+		strictEqual( checkbox.is( ":disabled" ), true,
+			"checkbox is disabled when inital option is set to true" );
+
+		checkbox.checkboxradio( "option", "disabled", false );
+
+		strictEqual( widget.hasClass( "ui-state-disabled" ), false,
+			"label has ui-state-disabled removed when disabled set to false" );
+		strictEqual( checkbox.is( ":disabled" ), false,
+			"checkbox has disabled prop removed when disabled set to false" );
+
+		checkbox.checkboxradio( "option", "disabled", true );
+
+		strictEqual( widget.hasClass( "ui-state-disabled" ), true,
+			"label gets ui-state-disabled when option set to true" );
+		strictEqual( checkbox.is( ":disabled" ), true,
+			"checkbox is disabled when option is set to true" );
+
+		checkbox.checkboxradio( "destroy" );
+		checkbox.prop( "disabled", true );
+		checkbox.checkboxradio();
+
+		strictEqual( widget.hasClass( "ui-state-disabled" ), true,
+			"label gets ui-state-disabled when checkbox is disabled in dom on startup" );
+		strictEqual( checkbox.is( ":disabled" ), true,
+			"checkbox is disabled when checkbox is disabled in dom on startup" );
+
+		checkbox.checkboxradio( "destroy" );
+		checkbox.checkboxradio({
+			disabled: null
+		});
+
+		strictEqual( widget.hasClass( "ui-state-disabled" ), true,
+			"passing null to disabled on startup checks the dom" );
+		strictEqual( checkbox.is( ":disabled" ), true,
+			"passing null to disabled on startup checks the dom" );
+	});
+	test( "options: icon", function() {
+		var checkbox = $( "#checkbox-option-icon" ),
+			widget;
+
+		expect( 9 );
+
+		checkbox.checkboxradio({ icon: false });
+
+		widget = checkbox.checkboxradio( "widget" );
+
+		strictEqual( widget.find( "span" ).length, 0,
+			"Label does not contain a span when created with icon:false" );
+
+		checkbox.checkboxradio( "destroy" );
+
+		checkbox.checkboxradio({
+			icon: true
+		});
+
+		strictEqual( widget.find( "span" ).length, 1,
+			"Label contains a span when created with icon:true" );
+		strictEqual( widget.find( "span" ).is( ".ui-checkboxradio-icon.ui-corner-all.ui-icon." +
+			"ui-icon-background.ui-icon-blank" ), true,
+			"Icon span has proper classes when created not checked" );
+
+		checkbox.checkboxradio( "destroy" ).prop( "checked", true );
+
+		checkbox.checkboxradio({
+			icon: true
+		});
+
+		strictEqual( widget.find( "span" ).is( ".ui-checkboxradio-icon.ui-corner-all.ui-icon." +
```

Comment:
This is the same list of classes as above. It gets repeated again below, though the one below has "ui-icon-blank" instead of "ui-icon-check". Could abstract this anyway, like this:

```js
var baseClasses = ".ui-checkboxradio-icon.ui-corner-all.ui-icon.ui-icon-background";
equal( widget.find( "span" ).attr( "class" ), baseClasses + ".ui-icon-check" );
// later
equal( widget.find( "span" ).attr( "class" ), baseClasses + ".ui-icon-blank" );
```